### PR TITLE
Issue#107: Added Embedding Youtube video in React Quill

### PR DESCRIPTION
**Pull request by dhairya-create**
This PR adds a custom video handler to Quill, allowing users to embed YouTube videos. It prompts users to input a YouTube URL and inserts the video in the cursor, enhancing content creation within the editor.

### CODE REVIEW

**Comment by aadeshkulkarni** [comment]
@dhairya-create  - Build fails. please check the error logs below

Screenshot 2024-05-27 at 11 23 03 PM

File: frontend/src/pages/Publish.tsx [hunk 1]
```diff
@@ -13,6 +13,22 @@ import GenerateAIBtn from '../components/GenerateAIBtn';
...
+
+// Register custom video handler with Quill
+const videoHandler = function (this: Quill) {
+  const range = this.getSelection();
+  const value = prompt('Please enter YouTube URL:');
+  if (value) {
+    this.insertEmbed(range.index, 'video', value, 'user');
+    this.setSelection(range.index + 1);
+  }
+};
```
Comment (aadeshkulkarni) [inline frontend/src/pages/Publish.tsx:25]: We can make videoHandler reusuable. 
Define it once in a util/lib file and use it in both files.

**Comment by aadeshkulkarni** [comment]
@dhairya-create - Please link the issue # to the PR.

File: frontend/src/layouts/EditPublishLayout.tsx [hunk 2]
```diff
@@ -25,6 +25,34 @@ const EditPublishLayout: React.FC<EditPublishLayoutProps> = ({
...
+        [{ 'list': 'ordered'}, { 'list': 'bullet' }, { 'list': 'check' }], 
+        ['bold', 'italic', 'underline', 'strike'],
+        [{ 'align': [] }],
+        [{ 'color': [] }, { 'background': [] }],
+        ['code-block'],
+        ['link', 'image', 'video'],
+        ['clean']
+      ],
+    },
+  };
```
Comment (aadeshkulkarni) [inline frontend/src/layouts/EditPublishLayout.tsx:54]: The modules should also be defined once, and reused in Publish & Edit, shouldn'it ?

File: frontend/src/layouts/EditPublishLayout.tsx [hunk 3]
```diff
@@ -1,11 +1,11 @@
 import React, { useRef } from 'react';
-import ReactQuill from 'react-quill';
+import ReactQuill,{Quill} from 'react-quill';
```
Comment (aadeshkulkarni) [inline frontend/src/layouts/EditPublishLayout.tsx:2]: linting issue. 
Please run the following commands once.
```bash
npm run lint-fix
npm run format
```

File: frontend/src/layouts/EditPublishLayout.tsx (hunk 2 above)
Comment (dhairya-create) [reply frontend/src/layouts/EditPublishLayout.tsx:54]: Yes thought the same thing, but was not able to find any file containing constants
Can you just tell me where all the constants for the front end are stored?

File: frontend/src/layouts/EditPublishLayout.tsx (hunk 3 above)
Comment (dhairya-create) [reply frontend/src/layouts/EditPublishLayout.tsx:2]: Have run this command, and haven't encountered any errors so far.

**Pull request merged**